recipes/default.rb

```diff
@@ -15,3 +15,30 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+include_recipe 'osl-docker'
+
+# Synapse management user
+user 'synapse-host' do
+  system true
+end
+
+# Synapse configuration, used within the docker container as a volume
+directory '/etc/synapse' do
+  owner 'synapse-host'
+  group 'synapse-host'
+  mode '750'
+end
+
+cookbook_file '/etc/synapse/homeserver.yaml' do
+  source 'homeserver.yaml'
+
+  owner 'synapse-host'
+  group 'synapse-host'
+end
+
+docker_container 'Matrix Synapse' do
+  repo 'matrixdotorg/synapse'
+  volumes ['/etc/synapse:/data']
+  env ["UID=#{node['ect']['passwd']['synapse-host']['uid']}", "GID=#{node['ect']['passwd']['synapse-host']['gid']}"]
```

Comment:
Please expand this array out i.e.:
```suggestion
  env [
    "UID=#{node['ect']['passwd']['synapse-host']['uid']}",
    "GID=#{node['ect']['passwd']['synapse-host']['gid']}"
  ]
```
Also, why the use of attributes here?